packages/dd-trace/src/profiling/exporters/agent.js

```diff
@@ -61,13 +66,22 @@ class AgentExporter {
         options.port = this._url.port
       }
 
+      this._logger.debug(`Submitting agent report to: ${JSON.stringify(options)}`)
+
       form.submit(options, (err, res) => {
-        if (err) return callback(err)
+        if (err || !res) return reject(err)
+
+        const chunks = []
+        res.on('data', chunk => chunks.push(chunk))
+        res.on('end', () => {
+          this._logger.debug(`Agent export response: ${Buffer.concat(chunks)}`)
```

Comment:
This should be in a callback to avoid the buffer when the logger is disabled.

Reply:
The logger only supports sync callbacks though. Maybe we could add support for async logging callbacks?

Reply:
Oh wait, you just mean the concat, right? I can do that. Async logging would still be cool too though.